drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/panel/IssuesSet.java

```diff
@@ -82,13 +82,14 @@ private int compareRowNumbers( final Set<Integer> rowNumbers,
                                 .filter( Optional::isPresent ).map( Optional::get ).collect( Collectors.toSet() ) );
 
         addAll( issues.stream()
-                      .filter( issue -> !MERGEABLE_ISSUES.contains( issue.getExplanationType() ) )
```

Comment:
If we invert the flow here, as in `issues.stream().filter(...).filter(...).foreach(add(...))`, no collection will ever need to be created - saving CPU cycles on creating the collection and then immediately GC'ing it.